src/Microsoft.Windows.ComputeVirtualization/Hcs.cs

```diff
@@ -0,0 +1,236 @@
+using System;
+using System.Runtime.InteropServices;
+
+[module: DefaultCharSet(CharSet.Unicode)]
+namespace Microsoft.Windows.ComputeVirtualization
+{
+    public class HcsFactory
+    {
+        public static IHcs GetHcs()
+        {
+            return new Hcs();
+        }
+
+        private class Hcs : IHcs
+        {
+            public bool EnumerateComputeSystems(string query, out string computeSystems)
```

Comment:
Not a huge deal but you could alternatively implement these explicitly `bool IHcs.EnumerateComputeSystems(...)`